### ISSUE TITLE
Any six digits as token

### ISSUE THREAD

**Issue by jvblue** [mricon/totp-cgi#31]
I have the totpcgi work and authenticating against LDAP. However while testing, I found I do not need to use the google otp as any six digit number will work. This has been created using http://vcdxorbust.com/totpcgi-and-freeradius-with-vmware-view/ 

What might need to be modified to accept only the google otp?

Thank you

**Comment by tykeal** [comment]
If the token is being used appropriately you'll see something like the following in /var/log/messages of your RADIUS system:

```
Mar 20 22:51:30 auth1 totp.cgi[22169]: Success: user=USERNAME, mode=PAM_SM_AUTH, host=xxx.xxx.xxx.xxx, message=Valid token used
```
If you aren't seeing a message like that at all, then you RADIUS isn't actually contacting topcgi and you likely misconfigured. You might want to try running RADIUS in debug mode. To do that, stop the currently running radiusd process and then launch it (as root) as follows:

```bash
radiusd -X
```

Now that radiusd is in debug mode, run your connection tests again. You'll find that you can trace the entire connection and see all the decoded (that is, plain text) passwords. So, either do this with a throw away account or be extra cautious with your password! ;)

**Comment by jvblue** [comment]
Looks like i didnt point the file to the correct location. TY. Now that i have that resolved i am getting an error.

 Reply-Message = "Denied access by rlm_perl authenticate function. ERROR: 500 Can't connect to totpcgi.abcd.net:8443 (SSL connect attempt failed with unknown errorerror:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed)"

Is this the local machine having an issue communicating to itself?

**Comment by tykeal** [comment]
That looks like the certificates you're using aren't trusted.

For totpcgi itself:
In your /etc/totpcgi/totpcgi.conf file make sure that in pincode_backend.ldap_cacert points to a pem file for the CA your LDAP system is using.

For RADIUS:
In your /etc/raddb/totpcgi-handler.yaml make sure that ca_file is pointing to a pem file for the CA that your totpcgi is using.

In one of our environments the LDAP CA is a different CA than the totpcgi configuration which just piggy-backs on our Puppet PKI.

Basically it boils down to this:

totpcgi _will_ validate that the CA the cert used by your LDAP is trusted.

The recommended apache configuration _requires_ a trusted client cert to connect to it

The RADIUS plugin _will_ validate that the totpcgi web URL is serving on a cert that is from a trusted CA.
